app/models/hostgroup.rb

```diff
@@ -206,6 +206,10 @@ def clone(name = "")
     new
   end
 
+  def hosts_count
+    Host::Managed.authorized.where(:hostgroup => id).size
```

Comment:
The permission needs to be specified as the first argument for `authorized`, otherwise any filter with resource type Host would be considered. In other words, this code would count in also hosts for which user does not have view permission but he/she has delete permission.